keps/sig-auth/3926-handling-undecryptable-resources/README.md

```diff
@@ -0,0 +1,933 @@
+
+If a single object of a resource type cannot be decrypted, listing resources of that
+type in a path prefix containing the object always fails, even if the rest of
+the resource instances is accessible.
+
+Currently, removing a resource that causes such failures is not possible.
+A cluster administrator must access etcd directly and remove the malformed data manually.
+
+This KEP proposes a way to identify resources that fail to decrypt or fail to be decoded
+into an object, and introduces a new delete option to ignore any storage checks in case
+such a read error occurs. This is done in order to be able to delete such a failing
+resource by using just Kubernetes API.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- provide a way to identify persisted resources that failed decryption or that cannot be
+  decoded
+- provide an option to delete a resource independently of its contents, if those
+  cannot be reached due to data transformation or data corruption
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- implementing system for ignoring different types of storage errors
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Improve resource retrieval errors to include more information about the object that
+failed transformation while it was being retrieved from the storage.
+
+Introduce a new `DeleteOption` that would allow deleting a resource even
+if we cannot retrieve its data.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+I accidentally removed my encryption key but only a few resources were encrypted
+with it. I know that these will either be recreated by a controller, or I can
+manually recreate them. I would like a simple way to figure out which resources
+fail decryption and I would like a way to remove them via Kubernetes API.
+
```

Comment:
Possibly a story 2 about wanting to delete a namespace but being blocked by encryption on a FooBar you no longer use(?)

Reply:
Thanks, that's a good idea. Added in the latest patch.